src/Compilers/Core/Portable/MetadataReader/MetadataHelpers.cs

```diff
@@ -507,13 +507,17 @@ private static short InferTypeArityFromMetadataName(string emittedTypeName, out
 
             // Given a name corresponding to <unmangledName>`<arity>,
             // extract the arity.
-            string stringRepresentingArity = emittedTypeName.Substring(indexOfManglingChar);
+            var aritySpan = emittedTypeName[indexOfManglingChar..];
 
             int arity;
-            bool nonNumericCharFound = !int.TryParse(stringRepresentingArity, NumberStyles.None, CultureInfo.InvariantCulture, out arity);
+#if NET
+            bool nonNumericCharFound = !int.TryParse(aritySpan, NumberStyles.None, CultureInfo.InvariantCulture, out arity);
+#else
+            bool nonNumericCharFound = !int.TryParse(aritySpan.ToString(), NumberStyles.None, CultureInfo.InvariantCulture, out arity);
+#endif
 
             if (nonNumericCharFound || arity < 0 || arity > short.MaxValue ||
-                stringRepresentingArity != arity.ToString())
+                !aritySpan.SequenceEqual(arity.ToString().AsSpan()))
```

Comment:
💡 Could also avoid this `ToString()` call for single-digit arity

Reply:
indeed!